agent/src/main/scala/za/co/absa/atum/agent/model/Measurement.scala

```diff
@@ -0,0 +1,130 @@
+package za.co.absa.atum.agent.model
+
+import org.apache.spark.sql.{Column, DataFrame, Dataset, Row}
+import org.apache.spark.sql.functions.col
+import org.apache.spark.sql.types.{DecimalType, LongType, StringType}
+import za.co.absa.atum.agent.core.MeasurementProcessor
+import org.apache.spark.sql.functions._
+import za.co.absa.atum.agent.core.MeasurementProcessor.MeasurementFunction
+import za.co.absa.atum.agent.model.Measurement.RecordCount
+import za.co.absa.spark.commons.implicits.StructTypeImplicits.StructTypeEnhancements
+
+/**
+ *  Type of different measures to be applied to the columns.
+ */
+trait Measurement extends MeasurementProcessor {
+
+  val controlCol: String
+}
+
+object Measurement {
+
+  case class RecordCount(
+    controlCol: String
+  ) extends Measurement {
+
+    override def function: MeasurementFunction =
+      (ds: Dataset[Row]) => ds.select(col(controlCol)).count().toString
```

Comment:
`Dataset[Row]` is the same as `DataFrame` and you are using `DataFrame` elsewhere, so why not to make it consistent?